tests/compile-test.rs

```diff
@@ -23,3 +23,23 @@ fn run_mode(mode: &'static str) {
 fn compile_test() {
     run_mode("compile-fail");
 }
+
+#[test]
+fn dogfood() {
```

Comment:
Could this test be placed in a separate file, not within the compile-fail test file?